### PR TITLE
Do not build LLL unless requested via the LLL cmake option

### PR DESCRIPTION
**Pull request by axic**
Part of #5393.

### CODE REVIEW

**Comment by axic** [comment]
This seemingly works on my computer, both with disabled and enabled.

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,9 +11,10 @@ eth_policy()
 set(PROJECT_VERSION "0.5.0")
 project(solidity VERSION ${PROJECT_VERSION})
 
+option(LLL "Build LLL" OFF)
 option(SOLC_LINK_STATIC "Link solc executable statically on supported platforms" OFF)
 option(LLLC_LINK_STATIC "Link lllc executable statically on supported platforms" OFF)
-option(INSTALL_LLLC "Include lllc executable in installation" OFF)
+option(INSTALL_LLLC "Include lllc executable in installation" LLL)
```
Comment (chfast) [inline CMakeLists.txt]:
```suggestion
option(INSTALL_LLLC "Include lllc executable in installation" ${LLL})
```

**Review by chfast — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (axic) [reply CMakeLists.txt]: Fixed.

**Review by chfast — DISMISSED** [review]

**Comment by axic** [comment]
This fails on tests. This change need to be more extensive. Better to skip it for 0.5.0.

**Comment by axic** [comment]
@chfast please review again, it required more extensive changes

File: docs/lll.rst [hunk 2]
```diff
@@ -9,6 +9,13 @@ LLL is a low-level language for the EVM with an s-expressions syntax.
 The Solidity repository contains an LLL compiler, which shares the assembler subsystem with Solidity.
 However, apart from maintaining that it still compiles, no other improvements are made to it.
 
+It is not built unless specifically requested:
+
+.. code-block:: bash
+
+    $ cmake -DLLL=ON ..
+    $ cmake --build
```
Comment (chfast) [inline docs/lll.rst]:
```suggestion
    $ cmake --build .
```

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -9,8 +9,10 @@ file(GLOB libevmasm_sources "libevmasm/*.cpp")
 file(GLOB libevmasm_headers "libevmasm/*.h")
 file(GLOB libyul_sources "libyul/*.cpp")
 file(GLOB libyul_headers "libyul/*.h")
+if (LLL)
 file(GLOB liblll_sources "liblll/*.cpp")
```
Comment (chfast) [inline test/CMakeLists.txt]:
```suggestion
    file(GLOB liblll_sources "liblll/*.cpp")
```

**Review by chfast — DISMISSED** [review]

**Comment by axic** [comment]
Rebased.

**Comment by axic** [comment]
@chfast @chriseth please review, should be ready now.

**Review by chfast — APPROVED** [review]

**Pull request merged**